modules/ROOT/pages/notifications/all-notifications.adoc

```diff
@@ -937,6 +938,240 @@ m|WARNING
+=== Neo.ClientNotification.Security.CommandHadNoEffect
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Code
+m|Neo.ClientNotification.Security.CommandHadNoEffect
+|Title
+a|`<command>` has no effect.
+|Severity
+m|INFORMATION
+|Category
+m|SECURITY
+|===
+
+The `<command>` in the title is the part of the command which had no effect.
+This could be either the full command given by the user or a subset of the given command.
+
+.Granting a role to a user who already have that role
+====
+Command::
++
+[source, cypher]
+----
+GRANT ROLE admin TO neo4j
+----
+
+Full title of the returned code::
+`GRANT ROLE admin TO neo4j` has no effect.
+
+Description of the returned code::
+The user already has the role. See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure this was the intended role and user.
+====
+
+.Revoking a role from a user who does not have that role
+====
+Command::
++
+[source, cypher]
+----
+REVOKE ROLE admin, reader FROM neo4j
+----
+
+Full title of the returned code::
+`REVOKE ROLE reader FROM neo4j` has no effect.
+
+Description of the returned code::
+The user does not have the role. See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure this was the intended role and user.
+====
+
+.Granting or denying a privilege to a role that already have that privilege
+====
+// This command will actually give 2 notifications, one for NODES and one for RELATIONSHIPS
+Command::
++
+[source, cypher]
+----
+GRANT MATCH {*} ON GRAPH * TO reader
+----
+
+Full title of the returned code::
+`GRANT TRAVERSE ON GRAPH * NODE * TO reader` has no effect.
+
+Description of the returned code::
+The role already has the privilege. See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure this was the intended privilege and role.
+====
+
+.Revoking a privilege from a role that does not have that privilege
+====
+Command::
++
+[source, cypher]
+----
+REVOKE WRITE ON GRAPH * FROM editor
+----
+
+Full title of the returned code::
+`REVOKE DENY WRITE ON GRAPH * FROM editor` has no effect.
+
+Description of the returned code::
+The role does not have the privilege. See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure this was the intended privilege and role.
+====
+
+[#_neo_clientnotification_security_impossiblerevokecommand]
+=== Neo.ClientNotification.Security.ImpossibleRevokeCommand
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Code
+m|Neo.ClientNotification.Security.ImpossibleRevokeCommand
+|Title
+a|`<command>` has no effect.
+|Severity
+m|WARNING
+|Category
+m|SECURITY
+|===
+
+The `<command>` in the title is the part of the command which had no effect.
+This could be either the full command given by the user or a subset of the given command.
+
+.Revoking a non-existing role from a user
+====
+Command::
++
+[source, cypher]
+----
+REVOKE ROLE Foo, reader FROM neo4j
```

Comment:
Here `Foo `is the role and `neo4j` is the user. Perhaps, we could stick to the same names.

Reply:
Not sure I follow, but I can look over the names again

Reply:
I mean that in the examples, sometimes Foo is a user, sometimes a role, and sometimes a graph.

Reply:
I think Foo is often the one that is non-existing which was the pattern I was trying to follow but I'll look it over next week hopefully

Reply:
fixed (https://github.com/neo4j/docs-status-codes/pull/54#discussion_r1193752367)